File: backend-python/app/services/attendance_service.py
import 'dart:convert';
import 'package:http/http.dart' as http;

class StudentAttendanceService {

  static Future<List<dynamic>>
      getAttendancePercentage() async {

    final response =
        await http.get(

      Uri.parse(
        "http://10.0.2.2:5000/attendance/percentage/1",
      ),
    );

    return jsonDecode(
      response.body,
    );
  }
}

File: frontend_flutter/lib/services/attendance_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'api_service.dart';

class AttendanceService {

  static Future<String>
      markAttendance({

    required String studentId,
    required String subject,
    required String status,
  }) async {

    final response =
        await http.post(

      ApiService.buildUri("/attendance/mark"),

      headers: {
        "Content-Type":
            "application/json",
      },

      body: jsonEncode({

        "student_id":
            int.parse(studentId),

        "subject":
            subject,

        "date":
            DateTime.now()
                .toString(),

        "status":
            status,
      }),
    );

    final data =
        jsonDecode(response.body);

    return data["message"];
  }
}